qemu/tests/serial_no_listening.py

```diff
@@ -0,0 +1,88 @@
+    """
+    Writing to a virtio serial port while no listening side:
+    1) boot guest with serial device.
+    2) guest send while no listening side.
+    3) host send while no listening side.
+    4) Transfer data between host and guest via serialport.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    def get_sender_cmd(sender):
+        """
+        Return host/guest send data cmd.
+
+        :param sender: who send the data, guest or host.
+        """
+
+        python_bin = '`command -v python python3 | head -1`'
+        session = vm.wait_for_login()
+        host_file_size, guest_file_size, host_action, guest_action\
+            = virtio_serial_file_transfer.get_command_options(sender, file_size)
+        host_file_name = virtio_serial_file_transfer.generate_data_file(
+                host_dir, host_file_size)
+        guest_file_name = virtio_serial_file_transfer.generate_data_file(
+                guest_dir, guest_file_size, session)
+        if sender == 'host':
+            sender_cmd = ("%s %s -t %s -s %s -f %s -a %s" % (python_bin,
+                          host_script, port_type, port_path, host_file_name,
+                          host_action))
+        else:
+            python_bin = params.get('python_bin', python_bin)
+            sender_cmd = ("%s %s -d %s -f %s -a %s" % (python_bin, guest_script,
+                          port_name, guest_file_name, guest_action))
+        session.close()
+        return sender_cmd
+
+    driver = params["driver_name"]
+    timeout = int(params.get("timeout", 60))
+    os_type = params["os_type"]
+    guest_path = params.get("guest_script_folder", "C:\\")
+    guest_scripts = params.get("guest_scripts",
+                               "VirtIoChannel_guest_send_receive.py")
+    port_name = params["file_transfer_serial_port"]
+    file_size = int(params.get("filesize", 10))
+    host_dir = data_dir.get_tmp_dir()
+    guest_dir = params.get("tmp_dir", '/var/tmp/')
+    host_script = params.get("host_script", "serial_host_send_receive.py")
+    host_script = os.path.join(data_dir.get_root_dir(), "shared", "deps",
+                               "serial", host_script)
+    guest_script = os.path.join(guest_path, params['guest_script'])
+
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+    if params["os_type"] == "windows":
+        session = utils_test.qemu.windrv_check_running_verifier(session, vm,
+                                                                test, driver)
+    virtio_serial_file_transfer.copy_scripts(guest_scripts, guest_path, vm)
+    port_type, port_path = virtio_serial_file_transfer.get_virtio_port_property(
+            vm, port_name)
+
+    # guest send while no listening side
+    try:
+        session.cmd_output(get_sender_cmd('guest'), timeout=timeout)
+    except aexpect.ShellTimeoutError:
+        pass
```

Comment:
Better have an `else` here, if you does not catch the exception, that's a fail, we don't expect that.